tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/graal/phases/TornadoHighTierContext.java

```diff
@@ -37,15 +37,19 @@ public class TornadoHighTierContext extends HighTierContext {
     protected final TaskMetaData meta;
     protected final boolean isKernel;
     private long batchThreads;
+    private int batchNumber;
+    private long batchSize;
 
     public TornadoHighTierContext(Providers providers, PhaseSuite<HighTierContext> graphBuilderSuite, OptimisticOptimizations optimisticOpts, ResolvedJavaMethod method, Object[] args,
-            TaskMetaData meta, boolean isKernel, long batchThreads) {
+            TaskMetaData meta, boolean isKernel, long batchThreads, int batchNumber, long batchSize) {
```

Comment:
Can we have a  `BatchConfig` class or maybe a record to pass around long batchThreads, int batchNumber, long batchSize?